### PR TITLE
[FLINK-28902][tests] rename FileSystemJobResultStoreTestInternal to FileSystemJobResultStoreTest and add Internal to test methods part

### PR DESCRIPTION
**Pull request by tonyzhu918**
## What is the purpose of the change

*rename FileSystemJobResultStoreTestInternal to FileSystemJobResultStoreTest and add Internal to test methods part*

## Brief change log

  - *rename FileSystemJobResultStoreTestInternal to FileSystemJobResultStoreTest and add Internal to test methods part*
  

## Verifying this change

Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

This change changed one test class for naming convetion and can be verified as follows:

  - *mvn clean verify*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by tonyzhu918** [comment]
@flinkbot run azure

File: flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/FileSystemJobResultStoreTest.java [hunk 1]
```diff
@@ -44,7 +44,7 @@
 
 /** Tests for the internal {@link FileSystemJobResultStore} mechanisms. */
 @ExtendWith(TestLoggerExtension.class)
-public class FileSystemJobResultStoreTestInternal {
```
Comment (XComp) [inline flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/FileSystemJobResultStoreTest.java:47]: We might want to come up with a more descriptive name for this test. The test is actually testing the file operations performed by the `FileSystemJobResultStore` implementation. The files function as some kind of user-facing interface as well. That makes it necessary to test this part.

What about renaming it into something like `FileSystemJobResultStoreFileOperationsTest`? WDYT?

File: flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/FileSystemJobResultStoreTest.java [hunk 2]
```diff
@@ -62,11 +62,11 @@ public void setupTest() throws IOException {
     }
 
     @Test
-    public void testValidEntryPathCreation() {
```
Comment (XComp) [inline flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/FileSystemJobResultStoreTest.java:65]: Renaming of the methods is not really necessary in the scope of FLINK-28902. The `Internal` suffix doesn't add any value to the method names, I feel. This applies to the other method name changes as well. 

The motivation around renaming the file is to make it being picked up by the surefire plugin. Therefore, renaming the class and the corresponding file is good enough. :+1:

**Review by XComp — CHANGES_REQUESTED** [review]
Thanks for providing this PR, @tonyzhu918 . I added a some comments below.

Additionally, please don't merge `master` into the branch but rather rebase your branch onto most recent master. That keeps the history in a better shape. :-)

**Comment by tonyzhu918** [comment]
Thanks for review. I created another PR https://github.com/apache/flink/pull/20845 .

**Comment by XComp** [comment]
@tonyzhu918 For the next time: You don't have to close PRs. Updating the corresponding branch is good enough. The benefit of using a single PR is to have all the discussion around code changes in one place.

**Comment by tonyzhu918** [comment]
@XComp got it and thanks!

**Pull request closed without merging**